src/main/java/net/dv8tion/jda/api/requests/restaction/GuildScheduledEventAction.java

```diff
@@ -0,0 +1,122 @@
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.OffsetDateTime;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+/**
+ * Extension of {@link net.dv8tion.jda.api.requests.RestAction RestAction} specifically
+ * designed to create a {@link GuildScheduledEvent GuildScheduledEvent}.
+ * This extension allows setting properties such as the name or description of an event before it is
+ * created.
+ *
+ * <p><b>Requirements</b><br>
+ * Events that are created are required to have a name, a location, and a start time. Depending on the
+ * type of location provided, an event will be of one of three different {@link GuildScheduledEvent.Type Types}:
+ * <ol>
+ *     <li>
+ *         {@link GuildScheduledEvent.Type#STAGE_INSTANCE Type.STAGE_INSTANCE}
+ *         <br>These events are set to take place inside of a {@link net.dv8tion.jda.api.entities.StageChannel StageChannel}. The
+ *         following permissions are required in the specified stage channel in order to create an event there:
+ *          <ul>
+ *              <li>{@link net.dv8tion.jda.api.Permission#MANAGE_EVENTS Permission.MANAGE_EVENTS}</li>
+ *              <li>{@link net.dv8tion.jda.api.Permission#MANAGE_CHANNEL Permission.MANAGE_CHANNEL}</li>
+ *              <li>{@link net.dv8tion.jda.api.Permission#VOICE_MUTE_OTHERS Permission.VOICE_MUTE_OTHERS}</li>
+ *              <li>{@link net.dv8tion.jda.api.Permission#VOICE_MOVE_OTHERS Permission.VOICE_MOVE_OTHERS}</li>
+ *         </ul>
+ *     </li>
+ *     <li>
+ *         {@link GuildScheduledEvent.Type#VOICE Type.VOICE}
+ *         <br>These events are set to take place inside of a {@link net.dv8tion.jda.api.entities.VoiceChannel}. The
+ *         following permissions are required in the specified voice channel in order to create an event there:
+ *         <ul>
+ *             <li>{@link net.dv8tion.jda.api.Permission#MANAGE_EVENTS Permission.MANAGE_EVENTS}</li>
+ *             <li>{@link net.dv8tion.jda.api.Permission#VIEW_CHANNEL Permission.VIEW_CHANNEL}</li>
+ *             <li>{@link net.dv8tion.jda.api.Permission#VOICE_CONNECT Permission.VOICE_CONNECT}</li>
+ *         </ul>
+ *     </li>
+ *     <li>
+ *         {@link GuildScheduledEvent.Type#EXTERNAL Type.EXTERNAL}
+ *         <br>These events are set to take place at an external location. {@link net.dv8tion.jda.api.Permission#MANAGE_EVENTS Permission.MANAGE_EVENTS}
+ *         is required on the guild level in order to create this type of event. Additionally, an end time <em>must</em>
+ *         also be specified.
+ *     </li>
+ * </ol>
+ *
+ * @see    net.dv8tion.jda.api.entities.Guild
+ * @see    Guild#createScheduledEvent(String, String, OffsetDateTime, OffsetDateTime) 
+ * @see    Guild#createScheduledEvent(String, GuildChannel, OffsetDateTime) 
+ */
+public interface GuildScheduledEventAction extends AuditableRestAction<GuildScheduledEvent>
```

Comment:
This is missing some setters like `setName`/`setStartTime`/`setEndTime`

Reply:
I had removed these because those attributes are now method paramaters. You figure they should still exist as setters ?

Reply:
Yes, I think the setters can still exist. It also simplifies checks.